### PR TITLE
Revert .spec.nodeName handling and add a default namespace selector

### PR DESCRIPTION
**Pull request by iholder101**
**What this PR does / why we need it**:
This PR changes how we handle .spec.nodeName.

All recent changes made in https://github.com/kubevirt/application-aware-quota/pull/28 and https://github.com/kubevirt/application-aware-quota/pull/47 will be reverted, and instead AAQ will start operating on in opt-in basis. The default namespace selector is now targeting namespaces with the `application-aware-quota/enable` label.

**Release note**:

```release-note
Revert .spec.nodeName handling and add a default namespace selector targeting application-aware-quota/enable
```

### CODE REVIEW

**Comment by Barakmor1** [comment]
It's very hard to follow all of these commits. Could you please squash the revert commits into one?

**Comment by iholder101** [comment]
> It's very hard to follow all of these commits. Could you please squash the revert commits into one?

Done, PTAL

File: tests/tests_suite_test.go [hunk 1]
```diff
@@ -125,6 +129,8 @@ func BuildTestSuite() {
 			nsList, _ := k8sClient.CoreV1().Namespaces().List(context.TODO(), metav1.ListOptions{LabelSelector: framework.NsPrefixLabel})
 			return nsList.Items
 		}, nsDeletedTimeout, pollInterval).Should(BeEmpty())
+
+		updateAAQNamespaceSelector(framework.ClientsInstance.AaqClient, nil)
```
Comment (Barakmor1) [inline tests/tests_suite_test.go:133]: can you please restore the original AAQ in the AfterSuite function similar to:
https://github.com/kubevirt/application-aware-quota/commit/593ac29d49917bd2e47ec3a5c1bdc80cbbf8e3b1#diff-0a9a24f44dfc4029d6c2465ca6246bf66829e97d236da6ff80172a8a9c73866a

Also we need to make sure that the namespace selector is propagated to the mutatingWebhookConfiguration before we start the tests

Comment (iholder101) [reply tests/tests_suite_test.go]: Good point!
Done :+1:

File: tests/tests_suite_test.go [hunk 2]
```diff
@@ -134,3 +152,46 @@ func BuildTestSuite() {
...
+
+		if reflect.DeepEqual(aaqWebhook.Webhooks[0].NamespaceSelector, originalWebhookSelector) {
+			return fmt.Errorf("expecting namespace selector to be updated")
+		}
+
+		return nil
+	}).WithTimeout(30*time.Second).WithPolling(time.Second).ShouldNot(HaveOccurred(), fmt.Sprintf("webhook namespace selector is not updated"))
+
+	return originalAAQ.Spec.NamespaceSelector, nil
+}
```
Comment (Barakmor1) [inline tests/tests_suite_test.go:197]: Can you please save the original AAQ outside of this function? I find it a bit confusing that this function returns the original AAQ.

File: tests/aaq_operator_test.go [hunk 3]
```diff
@@ -626,6 +626,69 @@ var _ = Describe("ALL Operator tests", func() {
...
+					return nil
+				}
+
+				if len(selector.MatchLabels) != 0 || len(selector.MatchExpressions) != 0 {
+					return fmt.Errorf("namespace selector should be empty. actual: %v", *mwc.Webhooks[0].NamespaceSelector)
+				}
+				
+				return nil
+			}).WithTimeout(90*time.Second).WithPolling(5*time.Second).ShouldNot(HaveOccurred(), "default namespace selector is not as expected")
+		})
```
Comment (Barakmor1) [inline tests/aaq_operator_test.go:691]: +1

File: tests/tests_suite_test.go (hunk 2 above)
Comment (iholder101) [reply tests/tests_suite_test.go]: Done!

**Comment by Barakmor1** [comment]
Thanks!

/lgtm
/approve
